src/test/server/query_handler_test.cpp

```diff
@@ -32,11 +33,17 @@ TEST_F(QueryHandlerTest, CreatePreparedPlan) {
 }
 
 TEST_F(QueryHandlerTest, BindParameters) {
-  QueryHandler::setup_prepared_plan("test_statement", "SELECT * FROM table_a WHERE a > ?");
-  const auto specification = PreparedStatementDetails{"test_statement", "", {123}};
+  QueryHandler::setup_prepared_plan("test_statement", "SELECT * FROM table_a WHERE a = ?");
+  const auto specification = PreparedStatementDetails{"test_statement", "", {12345}};
 
   const auto result = QueryHandler::bind_prepared_plan(specification);
-  EXPECT_EQ(result->type(), OperatorType::TableScan);
+  EXPECT_EQ(result->type(), OperatorType::Validate);
+
+  const auto get_table = std::dynamic_pointer_cast<const GetTable>(result->input_left()->input_left());
+  ASSERT_TRUE(get_table);
+
+  // Check that the optimizer was executed
+  ASSERT_FALSE(get_table->pruned_chunk_ids().empty());
```

Comment:
That seems a bit dirty but is probably the only way to test it without mocking?

Reply:
What part about this feels dirty to you? If there are chunks pruned, the optimizer was called.

Reply:
The part that we assume that the optimizer was called from a vector not being empty.

Reply:
We could run the test with a different parameter and check that a different chunk was pruned. Unless we expect any other component to change the pruned_chunk_ids, that should unambiguously show that the optimizer was executed.

Reply:
I don't think that we have to do that. The current solution is not too bad, it just seems a bit weird to assert that the optimizer was running based on a data structure that changed in some operator. It feels strange that we would have to adapt that test if we would disable the chunk pruning rule or that the test would fail if we adapted the predicate. But yes, there is probably no better way to do it right now.

Reply:
> It feels strange that we would have to adapt that test if we would disable the chunk pruning rule or that the test would fail if we adapted the predicate

Ok, agreed. I'll add a comment about that.